aten/src/ATen/native/cuda/ConvolutionMM2d.cu

```diff
@@ -414,19 +410,20 @@ std::tuple<Tensor&, Tensor&, Tensor&> slow_conv2d_backward_out_cuda(
     IntArrayRef kernel_size,
     IntArrayRef stride,
     IntArrayRef padding,
-    const Tensor& finput,
     Tensor& grad_input,
     Tensor& grad_weight,
     Tensor& grad_bias) {
   auto grad_output = grad_output_.expect_contiguous();
+
+  Tensor columns = at::empty({0}, self_.options());
   if (grad_input.defined()) {
     resize_output(grad_input, self_.sizes());
     auto weight = weight_.expect_contiguous();
 
     slow_conv2d_backward(
         self_, *grad_output,
         grad_input, *weight,
-        finput,
+        columns,
```

Comment:
I'm confused - we're creating an empty tensor `columns` and passing it directly to `slow_conv2d_backward` without computing the columns. Is this OK?

Reply:
It's okay because `slow_conv2d_backward` only uses the passed-in `columns` / `grad_columns` as an output tensor. It's a bit hard to read but here's where it's used.

```cpp
at::cuda::blas::gemm<scalar_t>(
    'n', 't',
    n, m, k,
    scalar_t(1),
    grad_output_n.data_ptr<scalar_t>(), n,
    weight.data_ptr<scalar_t>(), m,
    scalar_t(0),
    grad_columns.data_ptr<scalar_t>(), n
);
```

[gemm()](https://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-gemm) computes `C=αop(A)op(B)+βC`, with β=0 and C being `columns`, so the initial value of `columns` does not matter and it is used as an output.

Reply:
Thanks for the clarification!

The signature of `slow_conv2d_backward` should really have been the following to reflect the mutation...
```
void slow_conv2d_backward(
    const Tensor &input,
    const Tensor &grad_output,
    const Tensor &grad_input,
    const Tensor &weight_,
    Tensor &grad_columns,
    int kH, int kW,
    int dH, int dW,
    int padH, int padW)
```

Reply:
Also note that all the refactors in this stack pass all the gradcheck tests, so I think we're good in terms of backward correctness

Reply:
> 

Good point and I can change this if you'd like!